.github/workflows/update_golang_dependencies.yml

```diff
@@ -0,0 +1,56 @@
+name: Update Golang Dependencies
+
+on:
+  schedule:
+    - cron: "0 0 * * *" # Runs every day at midnight UTC
+  workflow_dispatch:
+
+permissions: read-all
+
+jobs:
+  update_golang_deps:
+    if: github.repository == 'vitessio/vitess'
+    permissions:
+      contents: write
+      pull-requests: write
+    name: Update Golang Dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.5
+
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Upgrade the Golang Dependencies
+        id: detect-and-update
+        run: |
+          go get -u ./...
+
+          output=$(git status -s)
+          if [ -z "${output}" ]; then
+            exit 0
+          fi
+
+          go mod tidy
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          branch: "upgrade-go-deps-on-main"
+          commit-message: "upgrade go deps"
+          signoff: true
+          delete-branch: true
+          title: "Upgrade the Golang Dependencies"
+          body: |
+            This Pull Request updates all the Goland dependencies to their latest version using `go get -u ./...`.
```

Comment:
shoot sorry i missed this

s/Goland/Golang

Reply:
Will be fixed with https://github.com/vitessio/vitess/pull/14896